[Swift in WebKit] Convert WTF from a library to a framework

rr-codes opened the pull request:
#### 4872ae5d32d504bb7e8c3dbebd5f44ddf42e69a9
<pre>
[Swift in WebKit] Convert WTF from a library to a framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=301814">https://bugs.webkit.org/show_bug.cgi?id=301814</a>
<a href="https://rdar.apple.com/163874240">rdar://163874240</a>

Reviewed by NOBODY (OOPS!).

Drafty draft draft

Test: Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
* Configurations/WebKitProjectPaths.xcconfig:
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/generate-unified-sources.sh:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/Info.plist: Added.
* Source/WTF/WTF.modulemap: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/WTF.xcodeproj/xcshareddata/xcschemes/WTF.xcscheme:
* Source/WTF/WTF_Private.modulemap: Added.
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/Scripts/generate-unified-sources.sh:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/CommandEncoder.swift:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-unified-sources.sh:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/scripts/generate-unified-sources.sh:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/lldb/lldbWebKitTester/Configurations/lldbWebKitTester.xcconfig:
* Tools/lldb/lldbWebKitTester/lldbWebKitTester.xcodeproj/project.pbxproj:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4872ae5d32d504bb7e8c3dbebd5f44ddf42e69a9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131540 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3872 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139047 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83323 "Failed to checkout and rebase branch from PR 53302") | ⏳ 🛠 ios-apple 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133410 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3897 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3712 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/139047 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/83323 "Failed to checkout and rebase branch from PR 53302") | ⏳ 🛠 mac-apple 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134486 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/155/builds/3897 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/139047 "Failed to checkout and rebase branch from PR 53302") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/155/builds/3897 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82240 "Failed to checkout and rebase branch from PR 53302") | | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/123565 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/155/builds/3897 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141694 "Failed to checkout and rebase branch from PR 53302") | | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/129994 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3616 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/141694 "Failed to checkout and rebase branch from PR 53302") | | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3664 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/156/builds/3712 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/141694 "Failed to checkout and rebase branch from PR 53302") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56804 "Failed to checkout and rebase branch from PR 53302") | | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3677 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/138/builds/42505 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/163011 "Failed to checkout and rebase branch from PR 53302") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3504 "Failed to checkout and rebase branch from PR 53302") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67088 "Failed to checkout and rebase branch from PR 53302") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/163011 "Failed to checkout and rebase branch from PR 53302") | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3759 "Failed to checkout and rebase branch from PR 53302") | | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3607 "Failed to checkout and rebase branch from PR 53302") | | | | 
<!--EWS-Status-Bubble-End-->

ian-twilightcoder commented on Source/WTF/WTF.xcodeproj/project.pbxproj:
```diff
@@ -4308,35 +4347,12 @@
 			);
 			outputPaths = (
-				"$(BUILT_PRODUCTS_DIR)/$(PRIVATE_HEADERS_FOLDER_PATH)/WTF.json",
+				/Users/richardrobinson/safari/OpenSource/WebKitBuild/Debug/WTF.framework/Versions/A/PrivateHeaders/WTF.json,
```
I think the previous path was correct?

ian-twilightcoder commented on Source/WTF/WTF.xcodeproj/project.pbxproj:
```diff
@@ -4196,8 +4235,8 @@
 			);
 			name = WTF;
```
I wonder if this should switch to "wtf" so that the module name can stay `wtf`. You'd have to make all of the headers public headers too.

ian-twilightcoder replied:
The other advantage of keeping with lowercase "wtf" is you don't have to update anyone's `#include`s.

ian-twilightcoder commented on Source/WTF/wtf_Private.modulemap:
Does Assertions.h, ExportMacros.h, SwiftBridging.h require C++? They're listed without any `requires` in the existing module

ian-twilightcoder commented on Source/WTF/wtf_Private.modulemap:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    explicit module spi {
+        umbrella "PrivateHeaders/spi"
```
Some of the spi/ headers are in the current `Cpp` submodule, are you saying that none of them actually require C++?

ian-twilightcoder replied:
Also I'm a little leery of nested umbrella directories but I guess if it works then ok...

rr-codes committed:
[Swift in WebKit] Convert WTF from a library to a framework
https://bugs.webkit.org/show_bug.cgi?id=301814
rdar://163874240

Reviewed by NOBODY (OOPS!).

Drafty draft draft

Test: Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
* Configurations/WebKitProjectPaths.xcconfig:
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/Configurations/JavaScriptCore.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/generate-unified-sources.sh:
* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/Info.plist: Added.
* Source/WTF/WTF.modulemap: Added.
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/WTF.xcodeproj/xcshareddata/xcschemes/WTF.xcscheme:
* Source/WTF/WTF_Private.modulemap: Added.
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/Scripts/generate-unified-sources.sh:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/CommandEncoder.swift:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-unified-sources.sh:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/scripts/generate-unified-sources.sh:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/lldb/lldbWebKitTester/Configurations/lldbWebKitTester.xcconfig:
* Tools/lldb/lldbWebKitTester/lldbWebKitTester.xcodeproj/project.pbxproj:
